tutor/src/components/student-dashboard/event-info-icon.cjsx

```diff
@@ -11,19 +11,22 @@ module.exports = React.createClass
     event: React.PropTypes.object.isRequired
 
   render: ->
-    due = moment(@props.event.due_at)
-    now = TimeStore.getNow()
+    {event} = @props
 
-    return null if @props.event.type isnt 'homework' or
-      @props.event.complete_exercise_count is @props.event.exercise_count or
-      due.isAfter(now, 'd')
+    now   = TimeStore.getNow()
+    dueAt = moment(event.due_at)
+    isIncomplete = event.complete_exercise_count isnt event.exercise_count
+    pastDue      = event.type is 'homework' and dueAt.isBefore(now, 'd')
+    workedLate   = moment(event.last_worked_at).isAfter(dueAt)
+
+    return null unless workedLate or (pastDue and isIncomplete)
 
     # use 'day' granularity for checking if the due date is today or after today
-    status = if due.isSame(now, 'd') then 'incomplete' else 'late'
+    status = if dueAt.isSame(now, 'd') then 'incomplete' else 'late'
```

Comment:
I wonder if we'll need to move from `day` granularity soon, since we do have due times now, but maybe another pr down the orad